backend/BalanceBuddy/src/test/java/com/example/balancebuddy/controllers/AuthControllerTest.java

```diff
@@ -0,0 +1,181 @@
+package com.example.balancebuddy.controllers;
+
+import com.example.balancebuddy.config.TokenGenerator;
+import com.example.balancebuddy.entities.Login;
+import com.example.balancebuddy.entities.SignUp;
+import com.example.balancebuddy.entities.MyUser;
+import com.example.balancebuddy.entities.Token;
+import com.example.balancebuddy.services.UserService;
+import io.jsonwebtoken.JwtException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.server.resource.authentication.BearerTokenAuthenticationToken;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationProvider;
+import org.springframework.security.web.authentication.logout.LogoutHandler;
+
+import java.util.Collections;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+class AuthControllerTest {
+
+    @InjectMocks
+    private AuthController authController;
+
+    @Mock
+    private UserService userService;
+
+    @Mock
+    private TokenGenerator tokenGenerator;
+
+    @Mock
+    private DaoAuthenticationProvider daoAuthenticationProvider;
+
+    @Mock
+    private JwtAuthenticationProvider refreshTokenAuthProvider;
+
+    @Mock
+    private LogoutHandler logoutHandler;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    void testRegister_Success() {
+        SignUp signupDTO = new SignUp("test@example.com", "John", "Doe", "Password123!");
+
+        MyUser user = new MyUser();
+        user.setEmail(signupDTO.getEmail());
+        user.setPassword(signupDTO.getPassword());
+
+        doNothing().when(userService).createUser(any(MyUser.class));
+
+        Authentication authentication = mock(Authentication.class);
+        Token mockToken = new Token();
+        when(tokenGenerator.createToken(any(Authentication.class))).thenReturn(mockToken);
+
+        ResponseEntity<?> responseEntity = authController.register(signupDTO);
+
+        assertEquals(200, responseEntity.getStatusCodeValue());
+    }
+
+    @Test
+    void testRegister_Fail() {
+        SignUp signupDTO = new SignUp("test@example.com", "John", "Doe", "wrong");
+
+
+        ResponseEntity<?> responseEntity = authController.register(signupDTO);

```

Comment:
Is it possible to replace the wildcard with the actual type of ResponseEntity object? If so, replace it in each occurrences.